tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/crossprotocol/AMQPResenderToOpenwireTest.java

```diff
@@ -0,0 +1,158 @@
+      super.setUp();
+      server = createServer(true, true);
+
+      factory = new ActiveMQConnectionFactory(OPENWIRE_URL);
+      qpidFactory = new JmsConnectionFactory(AMQP_URL);
+
+      Configuration serverConfig = server.getConfiguration();
+      serverConfig.getAddressesSettings().put("#", new AddressSettings().setAutoCreateQueues(true)
+            .setAutoCreateAddresses(true).setDeadLetterAddress(new SimpleString("ActiveMQ.DLQ")));
+      serverConfig.setSecurityEnabled(false);
+      server.start();
+
+      SimpleString coreQueueZero = new SimpleString(queueZeroName);
+      server.createQueue(coreQueueZero, RoutingType.ANYCAST, coreQueueZero, null, false, false);
+
+      SimpleString coreQueueOne = new SimpleString(queueOneName);
+      server.createQueue(coreQueueOne, RoutingType.ANYCAST, coreQueueOne, null, false, false);
+   }
+
+   @Override
+   @After
+   public void tearDown() throws Exception {
+      if (server != null) {
+         server.stop();
+         server = null;
+      }
+   }
+
+   @Test(timeout = 60_000)
+   public void openwireInternalBinaryPropertiesShouldBeResentAsByteArrays() throws Exception {
+      final List<Throwable> errors = new LinkedList<>();
+      final ExecutorService executor = Executors.newFixedThreadPool(2);
+      final CountDownLatch resenderReady = new CountDownLatch(1);
+
+      // AMQP resender
+      executor.execute(new Runnable() {
+         @Override
+         public void run() {
+            try {
+
+               Connection connection = qpidFactory.createConnection();
+               Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+               Queue queueZero = session.createQueue(queueZeroName);
+               Queue queueOne = session.createQueue(queueOneName);
+
+               MessageConsumer consumer = session.createConsumer(queueZero);
+               connection.start();
+               resenderReady.countDown();
+               Message message = consumer.receive();
+               assertNotNull(message);
+
+               MessageProducer producer = session.createProducer(queueOne);
+               producer.send(message);
+
+               connection.close();
+
+            } catch (Exception e) {
+               errors.add(e);
+            }
+         }
+      });
+
+      // OpenWire client
+      executor.execute(new Runnable() {
+         @Override
+         public void run() {
+            try {
+
+               Connection connection = factory.createConnection();
+               Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+               Queue queueZero = session.createQueue(queueZeroName);
+               Queue queueOne = session.createQueue(queueOneName);
+
+               MessageProducer producer = session.createProducer(queueZero);
+               Message testMessage = session.createTextMessage("test");
+               resenderReady.await(20, TimeUnit.SECONDS);
```

Comment:
Why do you need to wait for resenderReady?

Reply:
yes, I think it can be avoided, let me do some tests before removing it